examples/amm/src/contract.rs

```diff
@@ -124,15 +125,48 @@ impl Amm {
                 let input_pool_balance = self.get_pool_balance(input_token_idx).await?;
                 let output_pool_balance = self.get_pool_balance(output_token_idx).await?;
 
-                let output_amount = amm::calculate_output_amount(
+                let output_amount = self.calculate_output_amount(
                     input_amount,
                     input_pool_balance,
                     output_pool_balance,
                 )?;
 
-                self.receive_from_account(&owner, input_token_idx, input_amount)
+                self.receive_from_account(owner, input_token_idx, input_amount)
                     .await?;
-                self.send_to(&owner, output_token_idx, output_amount).await
+                self.send_to(owner, output_token_idx, output_amount).await?;
+
+                self.try_to_execute_swap_bids().await?;
+                Ok(())
+            }
+            OperationType::SwapBid {
+                owner,
+                input_token_idx,
+                input_amount,
+                minimum_output_token_amount,
+            } => {
+                if input_amount == Amount::ZERO || minimum_output_token_amount == Amount::ZERO {
```

Comment:
We could allow `minimum_output_token_amount == Amount::ZERO`: It just means I want to sell those tokens at the current price, whatever it is.

We could then remove the distinction between `SwapBid` and `Swap`: The bid is executed right away if the price allows it, otherwise it is stored. Just like in the matching engine example.

Reply:
Will address once we do swap bids